Lib/inspect.py

```diff
@@ -1976,7 +1990,9 @@ def _signature_fromstr(cls, obj, s, skip_bound_arg=True):
         module = None
 
     if not isinstance(module, ast.Module):
-        raise ValueError("{!r} builtin has invalid signature".format(obj))
+        exc = ValueError(f"invalid signature {text!r}")
```

Comment:
Commenting the if block on master and in this PR has no test failure so I guess this code was not tested. Perhaps a test could be added where `__text_signature__` has a invalid syntax as signature to see if `ValueError` is raised like below : 

```python
def test_python_function_invalid_text_signature(self):
    def func(*args, **kwargs):
        pass
    func.__text_signature__ = '($self, **kwargs, a)'
    with self.assertRaises(ValueError) as e:
        sig = inspect.signature(func)
```